### PR TITLE
allow `@tsconfig/node10` in `devDependencies`

### PR DESCRIPTION
**Pull request by JamieMagee**
**Task name**: N/A

**Description**: Update build scripts to allow `@tsconfig/node10` as an allowed dependency in `devDependencies`

**Documentation changes required:** N

**Added unit tests:** N

**Attached related issue:** #18118

**Checklist**:
- [ ] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

File: make-util.js [hunk 1]
```diff
@@ -159,11 +159,11 @@ var buildNodeTask = function (taskPath, outDir) {
     var overrideTscPath;
     if (test('-f', packageJsonPath)) {
         // verify no dev dependencies
-        // we allow a TS dev-dependency to indicate a task should use a different TS version
+        // we allow only two dev dependencies: typescript and @tsconfig/node10
         var packageJson = JSON.parse(fs.readFileSync(packageJsonPath).toString());
         var devDeps = packageJson.devDependencies ? Object.keys(packageJson.devDependencies).length : 0;
-        if (devDeps == 1 && packageJson.devDependencies["typescript"]) {
-            var version = packageJson.devDependencies["typescript"];
+        if (devDeps === 1 && packageJson.devDependencies['typescript'] || (devDeps === 2 && packageJson.devDependencies['typescript'] && packageJson.devDependencies['@tsconfig/node10'])) {
```
Comment (qianz2) [inline make-util.js:165]: What does this part do? When dev dependencies are 2, then we make sure it has both typescript and tsconfig?

Comment (JamieMagee) [reply make-util.js:165]: Yes. `devDependencies` must be either:
- `typescript`
- `typescript` and `@tsconfig/node10`

**Comment by JamieMagee** [comment]
@qianz2 Thanks for the approval. I don't have permissions to merge in this repo, so please merge if you think it's okay.

**Pull request merged**